app/controllers/deployments_controller.rb

```diff
@@ -48,10 +54,32 @@ def create
 private
 
   def application_by_repo
-    if (existing_app = Application.find_by(repo: repo_path))
-      existing_app
+    existing_apps = Application.where(repo: repo_path)
+    if existing_apps.present?
+      if existing_apps.length == 1
+        existing_apps[0]
+      else
+        flash[:alert] = format("Found multiple applications using repo: %<repo_path>s while using application_by_repo",
+                               repo_path: repo_path)
+        render :new
```

Comment:
I'm not entirely sure what'll happen with this here. I get what you're trying to do, but assuming this code runs, the `create` method is also going to run `head 200`. Maybe you could return something, and move this logic in to the `create` method, or perhaps raise an exception?

Reply:
fixed by returning nil if error occurred.

Reply:
Ok, I think that'll work.